### DISCUSSION TITLE
Question on getting ranges from Peloton classes

### DISCUSSION THREAD

**Discussion by lychee128** [cagnulein/qdomyos-zwift#395] [Q&A]
The target cadence and resistance displayed by class is great!  One question on the synchronization of the timing...

Are the target numbers displayed based on where I am in the class?  Or is it an estimate based on when the app discovers that I'm in a class?  

For example, if I pause for a couple of minutes after I've started a workout, will the target numbers also match where I am (after I restart from the pause)?

The ranges are sometimes off from what the instructor says but I understand this lag may be due to Peloton's own transcription.

Thanks.

**Comment by cagnulein** [comment]
Hi! I'm reading the metrics from the peloton API. The API gives me the resistance for each seconds. So from the start point on, QZ will follow this table but it doesn't know about your current status on peloton. This is why i added the peloton time tile in order to trim the delay that can occur if you press pause on peloton or in QZ.

**Reply by lychee128** [reply]
Got it.  

1) So if I am in the middle of a workout (eg, 2 minutes into a workout) and then start your app, does the API pick up the metrics as if it’s starting at the beginning of the workout?  So in this case, I should adjust the trim to be 120 secs (or however long my lag is)?

2) To minimize the lag, should I start your app right when I start a Peloton workout?


Thanks.

**Reply by cagnulein** [reply]
1. yes, you have to trim it to 120 seconds because peloton API doesn't give me the current point of your workout
2. you should press ok on the autofollow popup when the counter on peloton app is to 0:00  (after the intro)

**Comment by lychee128** [comment]
Thanks!
